### PR TITLE
Refactor specfile for building the worker package

### PR DESCRIPTION
**Pull request by r0x0d**
In this commit we are refactoring slightly how we build our worker, to make it output dependency information in the binary that will help QE to identify the versions we are using during the builds.

It also has some minor improvements to the Makefile, and with that, both the specfile and our development setup uses the same entrypoints for builds.

[HMS-3983](https://issues.redhat.com/browse/HMS-3983)

### CODE REVIEW

File: packaging/rhc-worker-script.spec [hunk 1]
```diff
@@ -44,38 +51,42 @@ managed by Red Hat Insights.
...
-# temporary directory in which the temporary script will be placed and executed.
+# Temporary directory in which the temporary script will be placed and executed.
 temporary_worker_directory: "/var/lib/rhc-worker-script"
-EOF
 
+# Pass environment variables to the script being executed
+# env:
+  # environment variables to be set for the script
+  # FOO: "some-string-value"
+  # BAR: "other-string-value"
```
Comment (andywaltlova) [inline packaging/rhc-worker-script.spec:85]: Is it commented on purpose?

Comment (andywaltlova) [reply packaging/rhc-worker-script.spec:85]: It's also not in README, I just noticed

File: Makefile [hunk 2]
```diff
@@ -10,44 +10,56 @@
...
-VENV ?= .venv3
-PRE_COMMIT ?= pre-commit
 GO_VERSION ?= 1.19
-CLIENT_ID ?= 00000000-0000-0000-0000-0000000000000
-SERVED_FILENAME ?= example_bash.yml
+
+BUILDFLAGS ?=
+LDFLAGS ?=
+ifeq ($(shell find . -name vendor), ./vendor)
+BUILDFLAGS += -mod=vendor
```
Comment (andywaltlova) [inline Makefile:22]: Is this the main change that makes it work?

**Review by andywaltlova — APPROVED** [review]
Honestly, this is black magic to me, I am not at all familiar with specfiles, I understand the changes, I see how rhc works, if you and QE team sees that this new way of building has benefits I am all for merge

File: Makefile (hunk 2 above)
Comment (andywaltlova) [reply Makefile:22]: Ah alright, with new build from this PR I am seeing now:

```
[root@vm-10-0-186-58 ~]# go version -m /usr/libexec/rhc/rhc-script-worker
/usr/libexec/rhc/rhc-script-worker: go1.20.12
	path	command-line-arguments
	dep	git.sr.ht/~spc/go-log	v0.1.1	
	dep	github.com/golang/protobuf	v1.5.3	
	dep	github.com/google/uuid	v1.6.0	
	dep	github.com/redhatinsights/yggdrasil	v0.0.0-20210811162724-41397343c25b	
	dep	golang.org/x/net	v0.23.0	
	dep	golang.org/x/sys	v0.18.0	
	dep	golang.org/x/text	v0.14.0	
	dep	google.golang.org/genproto/googleapis/rpc	v0.0.0-20230822172742-b8732ec3820d	
	dep	google.golang.org/grpc	v1.59.0	
	dep	google.golang.org/protobuf	v1.33.0	
	dep	gopkg.in/yaml.v3	v3.0.1	
	build	-buildmode=pie
	build	-compiler=gc
	build	-ldflags="-linkmode=external -compressdwarf=false -B 0x328b80c02072d97666206e224249b41795aebe27 -extldflags '-Wl,-z,relro '"
	build	CGO_ENABLED=1
	build	CGO_CFLAGS=
	build	CGO_CPPFLAGS="-D_FORTIFY_SOURCE=2 -fstack-protector-all"
	build	CGO_CXXFLAGS=
	build	CGO_LDFLAGS=
	build	GOARCH=amd64
	build	GOOS=linux
	build	GOAMD64=v1

```

Comment (r0x0d) [reply Makefile:22]: And this is how it looks like with the main build: 

```bash
[root@vm-10-0-184-120 ~]# scl enable go-toolset-1.19 'go version -m /usr/libexec/rhc/rhc-script-worker'
/usr/libexec/rhc/rhc-script-worker: go1.20.12
        path    rhc-script-worker-0.7
        build   -buildmode=pie
        build   -compiler=gc
        build   -ldflags=" -linkmode=external -B 0x0390eaffde65c3735698ff0a0bebbb62255fc0f4 -extldflags '-Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '"
        build   -tags=rpm_crashtraceback
        build   CGO_ENABLED=1
        build   CGO_CFLAGS=
        build   CGO_CPPFLAGS=
        build   CGO_CXXFLAGS=
        build   CGO_LDFLAGS=
        build   GOARCH=amd64
        build   GOOS=linux
        build   GOAMD64=v1
```

**Pull request merged**